### DISCUSSION TITLE
How do I leverage dynamic return types of a Prisma findFirst/findMany?

### DISCUSSION THREAD

**Discussion by samithjaganath** [prisma/prisma#24994] [Q&A]
### Question

I have the following structure in my rest service. 

CrudRepository.ts (A base interface that should be inherited by all models that would need to support crud)
```ts
export interface CrudRepository {
  getOne(filter: FindFirstFilter, tx?: Prisma.TransactionClient): Promise<WHAT_DO_I_RETURN_HERE>;
  getMany(filter: FindFirstFilter, tx?: Prisma.TransactionClient): Promise<WHAT_DO_I_RETURN_HERE[]>;
}
```

ICarRepository.ts (An interface for the Car repository. Car is a prisma model)
```ts
export interface ICarRepository extends CrudRepository {
  getOne(
    filter: Prisma.CarFindFirstArgs,
    tx?: Prisma.TransactionClient
  ): Promise<WHAT_DO_I_RETURN_HERE>;
}

```

CarRepository (Implementation class for ICarRepository)
```ts
@injectable()
export class CarRepository implements ICarRepository {
  constructor(
    @inject(Types.CarModelProvider) private carModelProvider: Factory<StorageModel<'Car'>>
  ) {}

  async getOne(
    filter: Prisma.CarFindFirstArgs,
    tx?: Prisma.TransactionClient
  ): Promise<WHAT_DO_I_RETURN_HERE> {
	return this.carModelProvider(tx).findFirst(filter);
  }
```

While I'm using CarRepository in a Service, I need it to **only return the keys mentioned in my select clause(with types)**. 

I project that this codebase could become large one day. Hence the high reusability targeted with repository functions. 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @samithjaganath 👋 

Can you please elaborate what you mean by: `I need it to only return the keys mentioned in my select clause(with types).`
Do you mean that you want to have a return type that only has a subset of fields and not all the fields of a model? Like what is described [here](https://www.prisma.io/docs/orm/prisma-client/type-safety/operating-against-partial-structures-of-model-types#problem-using-variations-of-the-generated-model-type)?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
